src/docs/compact.dox

```diff
@@ -2,11 +2,50 @@
 
 The WT_SESSION::compact method can be used to compact data sources.
 
-The data source does not have to be quiescent, compaction may be
-performed on an active data source.
+The data source does not have to be quiescent, compaction may be performed on an active data source.
 
-Because checkpoints named by the application are not discarded until
-explicitly removed or replaced, they may prevent WT_SESSION::compact
-from accomplishing anything.
+<b>Note:</b> Because checkpoints named by the application are not discarded until explicitly removed or replaced, they may prevent WT_SESSION::compact from accomplishing anything.
+
+@section foreground Foreground compaction
+
+Foreground compaction is executed when a specific URI is given to the WT_SESSION::compact method.
+The call returns once compaction has been executed on the specified URI.
+
+Example:
+@code
+# Execute compaction on "mytable" with the default parameters.
+int ret = session->compact(session, "table:mytable", NULL);
+
+# Execute compaction on "mytable" if at least 20MB of free space can be theoretically reclaimed by compaction.
+int ret = session->compact(session, "table:mytable", "free_space_target=20MB");
+@endcode
+
+@section background Background compaction
+
+The "background" configuration item can be used to interact with background compaction when calling
+the WT_SESSION::compact method. The call immediately returns to the application but does not
+guarantee compaction will be enabled/disabled at the same time. A successful call simply means the
```

Comment:
Could you elaborate on this part? 
```
A successful call simply means the request has been submitted to WiredTiger.
```

Reply:
Sure, see f0952709581d991ca1b5ccf6aad511487598677e